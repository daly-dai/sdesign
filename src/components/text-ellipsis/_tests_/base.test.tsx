import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';

import STextEllipsis from '../index';

import mountTest from '@daly/sdesign/tests/mountTest';

describe('STextEllipsis', () => {
  mountTest(STextEllipsis.Single);
  mountTest(() => <STextEllipsis.Single />);
  mountTest(STextEllipsis.Multi);
  mountTest(() => <STextEllipsis.Multi />);
  it('should render correctly', () => {
    const { container } = render(<STextEllipsis.Single />);
    expect(container).toMatchSnapshot();
  });
  it('should render Multi correctly', () => {
    const { container } = render(<STextEllipsis.Multi />);
    expect(container).toMatchSnapshot();
  });
});
