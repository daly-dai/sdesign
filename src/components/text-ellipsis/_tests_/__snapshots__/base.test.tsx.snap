// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`STextEllipsis > should render Multi correctly 1`] = `
<div>
  <div
    class="acss-hlsjwr"
    style="font-size: 14px;"
  />
</div>
`;

exports[`STextEllipsis > should render correctly 1`] = `
<div>
  <div
    class="acss-zifu8"
    style="font-size: 14px;"
  />
</div>
`;

exports[`STextEllipsis should render Multi correctly 1`] = `
<div>
  <div
    class="sd-ellipsis-multi  "
    style="font-size: 14px;"
  />
</div>
`;

exports[`STextEllipsis should render correctly 1`] = `
<div>
  <div
    class="sd-ellipsis-single  "
    style="font-size: 14px;"
  />
</div>
`;
